### PR TITLE
Cooperative Work Model

### PR DESCRIPTION
**Pull request by shawat94**
### Description:
This PR adds support for cooperative work on the task level, supplementing the existing support for challenge-level cooperative work types when submitting new challenges via line-delimited GeoJSONs containing task-specific cooperative work details.

Some additional changes include:
- Refactoring of constructions and setters for existing models to allow exception handling in the setter
- Consolidation of enums for future reusability

### Potential Impact:
Users will be able to pass cooperative work details into the task model when submitting tasks to MapRoulette. This will allow preliminary support for two types of cooperative work:

- Tag Fix (formerly 'QuickFix') actions which perform a simple operation via MapRoulette based on user approval (see [example task](https://maproulette.org/challenge/14452/task/66164468) generated with the model created by this PR)
- In-progress work submitted via an osmChange (.osc) file

### Functionality Test:
Task created using the 'setTags' methodology laid out in cooperative_work_example.py:
https://maproulette.org/challenge/14452/task/71543628

image

Task created using the 'Type 2' (in-progress edits generated from an OSMChange/.osc file):

https://maproulette.org/challenge/14452/task/75338558

.osc file contents:
`<osmChange version="0.6" generator="acme osm editor">
    <create>
        <node id="-1" changeset="-1" version="1" lat="55.69378277487475" lon="12.593096457148427" />
        <node id="-2" changeset="-1" version="1" lat="55.689697320811575" lon="12.597881450158676" />
        <way id="-3" changeset="-1" version="1">
            <nd ref="-1"/>
            <nd ref="-2"/>
        </way>
    </create>
</osmChange>`

JOSM view:
image

### Unit Test Approach:
Mock request assert and exception handling tests

### Test Results:
Passed

### CODE REVIEW

File: maproulette/models/challenge.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def medium_priority_rule(self):
         return self._medium_priority_rule
 
     @medium_priority_rule.setter
-    def high_priority_rule(self, value):
```
Comment (mattmanley) [inline maproulette/models/challenge.py:134]: It is strange to me that this diff is showing up - I don't see this issue in the dev code...

File: maproulette/models/cooperative_work.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+"""This module contains the definition of a Cooperative Work object in MapRoulette."""
+
+
+class CooperativeWorkModel:
+    """Definition for a Cooperative Work object"""
+
+    @property
+    def version(self):
+        """"""
```
Comment (mattmanley) [inline maproulette/models/cooperative_work.py:12]: Should we add some docstrings for these properties?

File: maproulette/models/project.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ def groups(self):
 
     @groups.setter
     def groups(self, value):
-        self._description = value
+        self._groups = value
```
Comment (mattmanley) [inline maproulette/models/project.py:47]: Good catch!

File: examples/cooperative_work_examples.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import maproulette
+import json
+import base64
```
[the remark this answers is not in the document]
Comment (shawat94) [reply examples/cooperative_work_examples.py:3]: Good catch, this import was used during initial testing for type 2 cooperative work which requires a base64-encoded changefile. I've updated the Coop work examples and PR description with results using these cooperative work examples just now, so this import is now used again.

File: maproulette/models/cooperative_work.py (hunk 2 above)
Comment (shawat94) [reply maproulette/models/cooperative_work.py]: Updated with docstrings

File: maproulette/models/challenge.py (hunk 1 above)
Comment (shawat94) [reply maproulette/models/challenge.py]: Looks like I hadn't merged changes from dev in quite a while! This diff should be gone now

**Review by mattmanley — APPROVED** [review]
I think this looks great. Thanks @shawat94 !

**Pull request merged**